libs/ml/tests/ml/ops/squeeze.cpp

```diff
@@ -0,0 +1,227 @@
+//------------------------------------------------------------------------------
+//
+//   Copyright 2018-2019 Fetch.AI Limited
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+//------------------------------------------------------------------------------
+
+#include "core/serializers/main_serializer_definition.hpp"
+#include "math/base_types.hpp"
+#include "math/tensor.hpp"
+#include "ml/core/graph.hpp"
+#include "ml/ops/placeholder.hpp"
+#include "ml/ops/squeeze.hpp"
+#include "ml/serializers/ml_types.hpp"
+#include "ml/utilities/graph_builder.hpp"
+#include "vectorise/fixed_point/fixed_point.hpp"
+
+#include "gtest/gtest.h"
+
+#include <cmath>
+#include <cstdint>
+#include <vector>
+
+template <typename T>
+class SqueezeTest : public ::testing::Test
+{
+};
+
+using BothTypes = ::testing::Types<fetch::math::Tensor<fetch::fixed_point::fp32_t>,
+                                   fetch::math::Tensor<fetch::fixed_point::fp64_t>,
+                                   fetch::math::Tensor<float>, fetch::math::Tensor<double>>;
+
+TYPED_TEST_CASE(SqueezeTest, BothTypes);
+
+TYPED_TEST(SqueezeTest, forward_test)
```

Comment:
I should have mentioned this before. It would be nice to extend this test slightly to demonstrate that squeeze actually removes the first 1 dimensional axis encountered starting with the trailing dimension.
So, if your original tensor is shape {1, 6, 1}; squeezing once gives {1, 6}, then squeezing again gives {6}, and squeezing a third time will throw.